SNLS_NLSPowell.h

```
@@ -0,0 +1,729 @@
+            return m_status;
+        }
+
+        while (m_status == unConverged && m_nIters < m_maxIter && m_fevals < m_maxIter) {
+            solveStep(residual, Jacobian, Qmat, diag, qtf, work_arr1, work_arr2, work_arr3);
+        }
+
+        return m_status;
+    }
+
+    // convenience wrapper, for the current _x
+    __snls_hdev__ 
+    inline bool computeRJ(double* const r,
+                          double* const J ) 
+    {
+        bool retval = this->m_crj.computeRJ(r, J, m_x);
+        // Add the finite difference portion of things back in here at a later point        
+        return retval;  
+    }
+
+    private:
+
+    __snls_hdev__
+    inline
+    SNLSStatus_t solveInit(double* const residual)
+    {
+        m_jevals = 0;
+        m_fevals = 1; 
+        m_status = snls::unConverged;
+        // at m_x
+        bool rjSuccess = this->computeRJ(residual, nullptr);
+        if ( !(rjSuccess) ) {
+            m_status = initEvalFailure;
+            return m_status;
+        }
+        
+        
+        m_res = snls::linalg::norm<m_nDim>(residual);
+        
+        // initialize iteration counter and monitors
+        m_nIters = 1;
+        m_ncsuc = 0;
+        m_ncfail = 0;
+        m_nslow1 = 0;
+        m_nslow2 = 0;
+        return m_status;
+    }
+
+    __snls_hdev__
+    inline
+    SNLSStatus_t solveStep(double* const residual,
+                           double* const Jacobian,
+                           double* const Q,
+                           double* const diag,
+                           double* const qtf,
+                           double* const work_arr1,
+                           double* const work_arr2,
+                           double* const work_arr3)
+    {
+        const double epsmch = snls::dbl_epsmch;
+        bool jeval = true;
+        
+        // at m_x
+        // We currently don't check on the success of this...
+        // and we should at this back in at some point
```

These comments have typos and are potentially confusing. 

If computeRJ fails then you should do at least something, rather than continuing merrily on your way.